lib/index.js

```diff
@@ -1,27 +1,41 @@
 const {accessSync, constants: {R_OK}} = require("fs");
-const {basename, dirname, join, resolve, relative} = require("path");
-const requireAll = require("require-all");
+const {dirname, join} = require("path");
 
 class Rule {
-  constructor(cfg, configFile, flags, fname) {
+  /**
+   * Creates a new linter Rule.
+   * @constructor
+   * @param  {object} cfg        The contents of the specified "manifest.json" or "package.json" file.
+   * @param  {string} configFile The fully qualified path of the specified config file (from `cfg` argument).
+   * @param  {object} flags      The CLI flags passed to the script (such as the "--manifest-path" or "--package-path").
+   * @param  {string} name       The name of the rule (such as "manifest/applications-gecko-id", or "package/addon-utils").
+   */
+  constructor(cfg, configFile, flags, name) {
     this.configPath = configFile;
     this.configDir = dirname(configFile);
     this.flags = flags;
-    this.name = relative(resolve(__dirname, ".."), fname).replace(/\.js$/, "");
-    this.label = basename(configFile);
-    const key = basename(configFile, ".json");
+    this.name = name;
+
+    const [key] = name.split("/");
     // where `key` is "manifest" or "package".
     this[key] = cfg;
+    this[`${key}Path`] = configFile;
```

Comment:
This look funny, but it's an odd convenience for doing stuff like `this.manifest` or `this.package` within rules, versus having to use `this.cfg`.
This line in particular also allows us to do `this.manifestPath` and `this.packageDir` instead of relying on more generic names.